content/applications/inventory_and_mrp/inventory/shipping_receiving/advanced_operations_shipping/print_on_validation.rst

```diff
@@ -156,3 +160,187 @@ be printed as:
+
+Assign printer
+--------------
+
+Refer to the :doc:`Connect a printer <../../../../general/iot/devices/printer>` documentation for
+details on connecting a printer to Odoo's *IoT* app. Upon completion, assign the carrier label to
+the printer, by navigating to :menuselection:`IoT app --> Devices`, and selecting the desired
+printer.
+
+.. image:: print_on_validation/select-printer.png
+   :align: center
+   :alt: Show a list of IoT devices.
+
+In the printer configuration form, go to the :guilabel:`Printer Reports` tab to configure the types
+of documents the printer automatically prints. Click :guilabel:`Add a line` to open the
+:guilabel:`Add: Reports` pop-up window. In the :guilabel:`Search...` bar, type `Shipping`, and
+select :guilabel:`Shipping Labels`.
+
+.. note::
+   The :guilabel:`Shipping Documents` report is for :ref:`export documents
+   <inventory/shipping_receiving/export-doc>`.
+
+.. image:: print_on_validation/printer-report.png
+   :align: center
+   :alt: Show carrier label report added to the *Printer Reports*.
+
+After adding the :guilabel:`Shipping Labels` report in the :guilabel:`Printer Reports` tab, ensure
+the :guilabel:`Report Type` matches the IoT-connected printer's type.
+
+- For laser printers, set the :guilabel:`Report Type` to :guilabel:`PDF`.
+- For Zebra printers, set the :guilabel:`Report Type` to :guilabel:`Text`.
+
+.. _inventory/shipping_receiving/label-type:
+
+Shipping carrier label type
+---------------------------
+
+Next, complete the setup for the :doc:`third-party shipping connector
+<../setup_configuration/third_party_shipper>`. After that, go to :menuselection:`Inventory app -->
+Configuration --> Shipping Methods`, and select the desired shipping method.
+
+On the shipping method configuration form, in the :guilabel:`[carrier name] Configuration` tab,
+ensure the :guilabel:`Label Format` matches the :ref:`report type assigned earlier
+<inventory/shipping_receiving/assign-printer>`:
+
+- For laser printers, set the :guilabel:`Label Format` to :guilabel:`PDF`.
+- For Zebra printers, set the :guilabel:`Label Format` to :guilabel:`ZPL2`.
+
+.. image:: print_on_validation/label-type.png
+   :align: center
+   :alt: Show the *Label Type* field on FedEx's shipping method configuration page.
+
+Example carrier label
+---------------------
+
+After validating the operation, the carrier label is generated in the chatter, and printed using the
+IoT-connected printer.
+
+.. seealso::
+   :doc:`Print carrier labels <labels>`
+
+.. image:: print_on_validation/carrier-label.png
+   :align: center
+   :alt: Show carrier label generated in the chatter.
+
+.. spoiler:: Example carrier label
+
+     .. figure:: print_on_validation/fedex-carrier-label.png
+        :align: center
+        :alt: Show an example carrier label for FedEx.
+
+        Carrier label for FedEx, containing the recipient address, tracking number, barcode, and
+        other shipping information.
```

Comment:
Since there are a large number of images in this document, I would like to minimize adding any more that we could possibly do without. I think the generated labels are very valuable to have included, but I think the `carrier-label.png` image can be removed
```suggestion
.. spoiler:: Example carrier label

     .. figure:: print_on_validation/fedex-carrier-label.png
        :align: center
        :alt: Show an example carrier label for FedEx.

        Carrier label for FedEx, containing the recipient address, tracking number, barcode, and
        other shipping information.

.. seealso::
   :doc:`Print carrier labels <labels>`
```